> @ronjouch regarding tests, I ran all of them, updated a couple of tests and added a couple to accommodate for possibly broken areas too. Do check [test.txt](https://github.com/ronjouch/whoarder/pull/2/files#diff-8a7cab4eaa51a22ff07f4a1c4dc223ddeaae0ec4ac40bd4759e88a0ab1e04f93R86)

@mynameismon nope, for me I still have 12/13 tests failing, all with your new error.

<details>
  <summary>Tests output on your branch</summary>

```
12:27:29 /o/whoarder (pr2|✔) python -m unittest
/opt/whoarder/whoarder/clippings.py:141: ResourceWarning: unclosed file <_io.BufferedReader name='/opt/whoarder/whoarder/test/test.txt'>
  rawdata = open(source, "rb").read()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
E/usr/lib/python3.9/unittest/case.py:621: ResourceWarning: unclosed file <_io.TextIOWrapper name='/opt/whoarder/whoarder/test/test.txt' mode='r' encoding='utf-8-sig'>
  outcome.errors.clear()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
EEEEEEEEEEE.
======================================================================
ERROR: test_bom_stripped (whoarder.test.TestImport)
Ensure repeated BOMs incorrectly written by Kindle are stripped
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/whoarder/whoarder/test/__init__.py", line 10, in setUp
    clippingsimporter = Clippings(path)
  File "/opt/whoarder/whoarder/clippings.py", line 17, in __init__
    self._fetch()
  File "/opt/whoarder/whoarder/clippings.py", line 34, in _fetch
    for clipping in clippings:
  File "/opt/whoarder/whoarder/clippings.py", line 100, in __next__
    raise InvalidFormatException('''Input file doesn't seem to be
whoarder.clippings.InvalidFormatException: Input file doesn't seem to be
                    a clippings file, separators are missing or damaged

======================================================================
ERROR: test_count (whoarder.test.TestImport)
test.txt should yield a certain number of clippings.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/whoarder/whoarder/test/__init__.py", line 10, in setUp
    clippingsimporter = Clippings(path)
  File "/opt/whoarder/whoarder/clippings.py", line 17, in __init__
    self._fetch()
  File "/opt/whoarder/whoarder/clippings.py", line 34, in _fetch
    for clipping in clippings:
  File "/opt/whoarder/whoarder/clippings.py", line 100, in __next__
    raise InvalidFormatException('''Input file doesn't seem to be
whoarder.clippings.InvalidFormatException: Input file doesn't seem to be
                    a clippings file, separators are missing or damaged

======================================================================
ERROR: test_count_bookmarks (whoarder.test.TestImport)
test.txt should yield 2 bookmarks
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/whoarder/whoarder/test/__init__.py", line 10, in setUp
    clippingsimporter = Clippings(path)
  File "/opt/whoarder/whoarder/clippings.py", line 17, in __init__
    self._fetch()
  File "/opt/whoarder/whoarder/clippings.py", line 34, in _fetch
    for clipping in clippings:
  File "/opt/whoarder/whoarder/clippings.py", line 100, in __next__
    raise InvalidFormatException('''Input file doesn't seem to be
whoarder.clippings.InvalidFormatException: Input file doesn't seem to be
                    a clippings file, separators are missing or damaged

======================================================================
ERROR: test_count_highlights (whoarder.test.TestImport)
test.txt should yield 13 highlights
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/whoarder/whoarder/test/__init__.py", line 10, in setUp
    clippingsimporter = Clippings(path)
  File "/opt/whoarder/whoarder/clippings.py", line 17, in __init__
    self._fetch()
  File "/opt/whoarder/whoarder/clippings.py", line 34, in _fetch
    for clipping in clippings:
  File "/opt/whoarder/whoarder/clippings.py", line 100, in __next__
    raise InvalidFormatException('''Input file doesn't seem to be
whoarder.clippings.InvalidFormatException: Input file doesn't seem to be
                    a clippings file, separators are missing or damaged

======================================================================
ERROR: test_count_notes (whoarder.test.TestImport)
test.txt should yield 1 note
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/whoarder/whoarder/test/__init__.py", line 10, in setUp
    clippingsimporter = Clippings(path)
  File "/opt/whoarder/whoarder/clippings.py", line 17, in __init__
    self._fetch()
  File "/opt/whoarder/whoarder/clippings.py", line 34, in _fetch
    for clipping in clippings:
  File "/opt/whoarder/whoarder/clippings.py", line 100, in __next__
    raise InvalidFormatException('''Input file doesn't seem to be
whoarder.clippings.InvalidFormatException: Input file doesn't seem to be
                    a clippings file, separators are missing or damaged

======================================================================
ERROR: test_first_clipping (whoarder.test.TestImport)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/whoarder/whoarder/test/__init__.py", line 10, in setUp
    clippingsimporter = Clippings(path)
  File "/opt/whoarder/whoarder/clippings.py", line 17, in __init__
    self._fetch()
  File "/opt/whoarder/whoarder/clippings.py", line 34, in _fetch
    for clipping in clippings:
  File "/opt/whoarder/whoarder/clippings.py", line 100, in __next__
    raise InvalidFormatException('''Input file doesn't seem to be
whoarder.clippings.InvalidFormatException: Input file doesn't seem to be
                    a clippings file, separators are missing or damaged

======================================================================
ERROR: test_multi_line_highlights (whoarder.test.TestImport)
Multiline clippings should be read correctly
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/whoarder/whoarder/test/__init__.py", line 10, in setUp
    clippingsimporter = Clippings(path)
  File "/opt/whoarder/whoarder/clippings.py", line 17, in __init__
    self._fetch()
  File "/opt/whoarder/whoarder/clippings.py", line 34, in _fetch
    for clipping in clippings:
  File "/opt/whoarder/whoarder/clippings.py", line 100, in __next__
    raise InvalidFormatException('''Input file doesn't seem to be
whoarder.clippings.InvalidFormatException: Input file doesn't seem to be
                    a clippings file, separators are missing or damaged

======================================================================
ERROR: test_presence_author (whoarder.test.TestImport)
Each clipping should reference its author's name.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/whoarder/whoarder/test/__init__.py", line 10, in setUp
    clippingsimporter = Clippings(path)
  File "/opt/whoarder/whoarder/clippings.py", line 17, in __init__
    self._fetch()
  File "/opt/whoarder/whoarder/clippings.py", line 34, in _fetch
    for clipping in clippings:
  File "/opt/whoarder/whoarder/clippings.py", line 100, in __next__
    raise InvalidFormatException('''Input file doesn't seem to be
whoarder.clippings.InvalidFormatException: Input file doesn't seem to be
                    a clippings file, separators are missing or damaged

======================================================================
ERROR: test_presence_book (whoarder.test.TestImport)
Each clipping should reference a book.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/whoarder/whoarder/test/__init__.py", line 10, in setUp
    clippingsimporter = Clippings(path)
  File "/opt/whoarder/whoarder/clippings.py", line 17, in __init__
    self._fetch()
  File "/opt/whoarder/whoarder/clippings.py", line 34, in _fetch
    for clipping in clippings:
  File "/opt/whoarder/whoarder/clippings.py", line 100, in __next__
    raise InvalidFormatException('''Input file doesn't seem to be
whoarder.clippings.InvalidFormatException: Input file doesn't seem to be
                    a clippings file, separators are missing or damaged

======================================================================
ERROR: test_presence_contents (whoarder.test.TestImport)
Each clipping should have contents.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/whoarder/whoarder/test/__init__.py", line 10, in setUp
    clippingsimporter = Clippings(path)
  File "/opt/whoarder/whoarder/clippings.py", line 17, in __init__
    self._fetch()
  File "/opt/whoarder/whoarder/clippings.py", line 34, in _fetch
    for clipping in clippings:
  File "/opt/whoarder/whoarder/clippings.py", line 100, in __next__
    raise InvalidFormatException('''Input file doesn't seem to be
whoarder.clippings.InvalidFormatException: Input file doesn't seem to be
                    a clippings file, separators are missing or damaged

======================================================================
ERROR: test_presence_date (whoarder.test.TestImport)
Each clipping should have a date.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/whoarder/whoarder/test/__init__.py", line 10, in setUp
    clippingsimporter = Clippings(path)
  File "/opt/whoarder/whoarder/clippings.py", line 17, in __init__
    self._fetch()
  File "/opt/whoarder/whoarder/clippings.py", line 34, in _fetch
    for clipping in clippings:
  File "/opt/whoarder/whoarder/clippings.py", line 100, in __next__
    raise InvalidFormatException('''Input file doesn't seem to be
whoarder.clippings.InvalidFormatException: Input file doesn't seem to be
                    a clippings file, separators are missing or damaged

======================================================================
ERROR: test_presence_location (whoarder.test.TestImport)
Each clipping should reference the kindle location it appeared on.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/whoarder/whoarder/test/__init__.py", line 10, in setUp
    clippingsimporter = Clippings(path)
  File "/opt/whoarder/whoarder/clippings.py", line 17, in __init__
    self._fetch()
  File "/opt/whoarder/whoarder/clippings.py", line 34, in _fetch
    for clipping in clippings:
  File "/opt/whoarder/whoarder/clippings.py", line 100, in __next__
    raise InvalidFormatException('''Input file doesn't seem to be
whoarder.clippings.InvalidFormatException: Input file doesn't seem to be
                    a clippings file, separators are missing or damaged

----------------------------------------------------------------------
Ran 13 tests in 0.010s

FAILED (errors=12)
```
</details>

My python is rusty and I may be doing something wrong. But still, tests do run fine on `master`. 

<details>
  <summary>Tests output on <code>master</code></summary>

```
12:30:10 /o/whoarder (master|✔) python -m unittest
/opt/whoarder/whoarder/clippings.py:139: ResourceWarning: unclosed file <_io.BufferedReader name='/opt/whoarder/whoarder/test/test.txt'>
  rawdata = open(source, "rb").read()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
Failed to import the following note, please report to https://github.com/ronjouch/whoarder :
  ['Bogus Highlight', 'nonsense', '', 'This Highlight should fail to import and log an error message']

.Failed to import the following note, please report to https://github.com/ronjouch/whoarder :
  ['Bogus Highlight', 'nonsense', '', 'This Highlight should fail to import and log an error message']

.Failed to import the following note, please report to https://github.com/ronjouch/whoarder :
  ['Bogus Highlight', 'nonsense', '', 'This Highlight should fail to import and log an error message']

.Failed to import the following note, please report to https://github.com/ronjouch/whoarder :
  ['Bogus Highlight', 'nonsense', '', 'This Highlight should fail to import and log an error message']

.Failed to import the following note, please report to https://github.com/ronjouch/whoarder :
  ['Bogus Highlight', 'nonsense', '', 'This Highlight should fail to import and log an error message']

[{'book': '_BOOK_', 'author': '_AUTHOR_LAST_NAME_, _AUTHOR_FIRST_NAME_', 'type': '_TYPE_', 'page': 'Page 42', 'location': '123-321', 'date': '_DATE_', 'contents': 'contents'}, {'book': 'Title With a Long Name', 'author': 'Author 1 Name, Author 1 First Name', 'type': 'Highlight', 'page': 'Page 7', 'location': '117-119', 'date': 'Tuesday, April 9, 2013 7:21:15 PM', 'contents': '“Remarkable quote about stuff,” said John P., “especially now.”'}, {'book': 'Title With a Long Name', 'author': 'Author 1 Name, Author 1 First Name', 'type': 'Note', 'page': 'Unnumbered Page', 'location': '118', 'date': 'Tuesday, April 9, 2013 7:23:45 PM', 'contents': 'See Le Petit Prince'}, {'book': 'Title With a Long Name', 'author': 'Author 1 Name, Author 1 First Name', 'type': 'Highlight', 'page': 'Unnumbered Page', 'location': '218-219', 'date': 'Tuesday, April 9, 2013 7:37:21 PM', 'contents': 'John Doe remarked: “Man, yeah.”'}, {'book': 'Title With a Long Name', 'author': 'Author 1 Name, Author 1 First Name', 'type': 'Highlight', 'page': 'Page 8', 'location': '348-352', 'date': 'Tuesday, April 9, 2013 8:21:06 PM', 'contents': 'Strange characters like @!#$!(&$)%!($*#)($#)@!&$()#!@*$)!@&$!@)*#$&@(!éfèôâfüwej)'}, {'book': 'Walden', 'author': 'Thoreau, Henry David', 'type': 'Highlight', 'page': 'Page 3', 'location': '45-48', 'date': 'Wednesday, April 17, 2013 7:45:27 AM', 'contents': 'Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor. Ramblings on labor.'}, {'book': 'Walden', 'author': 'Thoreau, Henry David', 'type': 'Highlight', 'page': 'Page 3', 'location': '52-53', 'date': 'Wednesday, April 17, 2013 7:46:02 AM', 'contents': 'Ramblings on nature. Ramblings on nature. Ramblings on nature. Ramblings on nature. Ramblings on nature. Ramblings on nature. Ramblings on nature. Ramblings on nature. Ramblings on nature. Ramblings on nature.'}, {'book': 'Walden', 'author': 'Thoreau, Henry David', 'type': 'Highlight', 'page': 'Page 3', 'location': '52-55', 'date': 'Wednesday, April 17, 2013 7:47:22 AM', 'contents': 'Market makes us into machines. Market makes us into machines. Market makes us into machines. Market makes us into machines. Market makes us into machines. Market makes us into machines. Market makes us into machines. Market makes us into machines. Market makes us into machines. Market makes us into machines.'}, {'book': 'The Robert C. Martin Clean Code Collection (Collection)', 'author': 'Martin, Robert C.', 'type': 'Highlight', 'page': 'Page 27', 'location': '404-404', 'date': 'Saturday, April 27, 2013 5:29:07 PM', 'contents': 'Uncle bob telling us that practice is important.'}, {'book': 'The Robert C. Martin Clean Code Collection (Collection)', 'author': 'Martin, Robert C.', 'type': 'Highlight', 'page': 'Page 27', 'location': '403-406', 'date': 'Saturday, April 27, 2013 5:29:35 PM', 'contents': 'Uncle bob telling us about detail detail detail detail detail detail detail detail, and the broken window metaphor'}, {'book': 'The Robert C. Martin Clean Code Collection (Collection)', 'author': 'Martin, Robert C.', 'type': 'Highlight', 'page': 'Page 29', 'location': '435-435', 'date': 'Saturday, April 27, 2013 5:33:42 PM', 'contents': 'Uncle bob telling us about readability.'}, {'book': 'The Robert C. Martin Clean Code Collection (Collection)', 'author': 'Martin, Robert C.', 'type': 'Highlight', 'page': 'Page 31', 'location': '462-464', 'date': 'Saturday, April 27, 2013 5:38:22 PM', 'contents': 'Uncle bob telling us about indentation style and bug density.'}, {'book': 'Essays from the Nick of Time: Reflections and Refutations', 'author': 'Slouka, Mark', 'type': 'Highlight', 'page': 'Page 6', 'location': '128-131', 'date': 'Sunday, May 19, 2013 1:32:46 AM', 'contents': 'Slouka on progress and numbers.'}, {'book': 'Boneshaker', 'author': 'Priest, Cherie', 'type': 'Bookmark', 'page': 'Unnumbered Page', 'location': '441', 'date': 'Monday, December 30, 2013 6:59:23 PM', 'contents': ''}, {'book': 'Your Kindle is updated!', 'author': 'Amazon', 'type': 'Bookmark', 'page': 'Unnumbered Page', 'location': '10', 'date': 'Tuesday, March 11, 2014 7:20:18 PM', 'contents': ''}, {'book': 'Essays from the Nick of Time: Reflections and Refutations', 'author': 'Slouka, Mark', 'type': 'Highlight', 'page': 'Page 20', 'location': '296-296', 'date': 'Sunday, May 19, 2013 2:01:07 AM', 'contents': 'Slouka talking about understanding and shaping and understanding and shaping and understanding and shaping and understanding and shaping and understanding and shaping and understanding and shaping.'}, {'book': 'Essays from the Nick of Time: Reflections and Refutations', 'author': 'Slouka, Mark', 'type': 'Highlight', 'page': 'Page 40', 'location': '557-558', 'date': 'Monday, May 20, 2013 12:02:43 AM', 'contents': 'Slouka quoting Thoreau about margins to life and space and silence and the forming of self and space and silence and the forming of self and space and silence and the forming of self.'}]
.Failed to import the following note, please report to https://github.com/ronjouch/whoarder :
  ['Bogus Highlight', 'nonsense', '', 'This Highlight should fail to import and log an error message']

.Failed to import the following note, please report to https://github.com/ronjouch/whoarder :
  ['Bogus Highlight', 'nonsense', '', 'This Highlight should fail to import and log an error message']

.Failed to import the following note, please report to https://github.com/ronjouch/whoarder :
  ['Bogus Highlight', 'nonsense', '', 'This Highlight should fail to import and log an error message']

.Failed to import the following note, please report to https://github.com/ronjouch/whoarder :
  ['Bogus Highlight', 'nonsense', '', 'This Highlight should fail to import and log an error message']

.Failed to import the following note, please report to https://github.com/ronjouch/whoarder :
  ['Bogus Highlight', 'nonsense', '', 'This Highlight should fail to import and log an error message']

.Failed to import the following note, please report to https://github.com/ronjouch/whoarder :
  ['Bogus Highlight', 'nonsense', '', 'This Highlight should fail to import and log an error message']

..
----------------------------------------------------------------------
Ran 12 tests in 0.007s

OK
```
</details>

Does tests actually pass for you? If yes, halp welcome with my setup.

--- a/whoarder/clippings.py
+++ b/whoarder/clippings.py
@@ -150,5 +150,5 @@ def _detect_encoding(source):
     return detected_encoding
 
 
-class InvalidFormatException(Exception):
+class InvalidFormatException(BaseException):
     pass

--- a/whoarder/test/__init__.py
+++ b/whoarder/test/__init__.py
@@ -1,4 +1,4 @@
-from whoarder.clippings import Clippings
+from whoarder.clippings import Clippings, InvalidFormatException
 import unittest
 import os
 
@@ -106,6 +106,22 @@ class TestWrongImport(unittest.TestCase):
         '''
         with self.assertRaises(FileNotFoundError):
             self.clippings = Clippings('bar.txt')
+
+
+class TestMulitlineHighlights(unittest.TestCase):
+
+    def setUp(self):
+        self.path = (os.path.dirname(__file__)) + "/failed_test.txt"
+
+    def test_failed_multiline_highlights(self):
+        '''
+        the multiline highlight for failed_test.txt should return an InvalidFormatException
+        
+        '''
+        with self.assertRaises(InvalidFormatException):
+            Clippings(self.path)
+        # self.assertIsNone(None)
+    
 
 if __name__ == '__main__':
     unittest.main()

--- a/whoarder/test/failed_test.txt
+++ b/whoarder/test/failed_test.txt
@@ -1,0 +1,13 @@
+﻿Multiline Failed Highlight (Author 1 Name, Author 1 First Name)
+- Your Highlight on Unnumbered Page | Location 218-219 | Added on Tuesday, April 9, 2013 7:37:21 PM
+
+John Doe remarked: "Man, yeah.
+
+This is tricky stuff"
+﻿Multiline Highlight (Author 1 Name, Author 1 First Name)
+- Your Highlight on Unnumbered Page | Location 218-219 | Added on Tuesday, April 9, 2013 7:37:21 PM
+
+John Doe remarked: "Man, yeah.
+
+This is tricky stuff"
+==========

--- a/whoarder/test/test.txt
+++ b/whoarder/test/test.txt
@@ -90,19 +90,6 @@ John Doe remarked: "Man, yeah.
 
 This is tricky stuff"
 ==========
-﻿Multiline Failed Highlight (Author 1 Name, Author 1 First Name)
-- Your Highlight on Unnumbered Page | Location 218-219 | Added on Tuesday, April 9, 2013 7:37:21 PM
-
-John Doe remarked: "Man, yeah.
-
-This is tricky stuff"
-﻿Multiline Highlight (Author 1 Name, Author 1 First Name)
-- Your Highlight on Unnumbered Page | Location 218-219 | Added on Tuesday, April 9, 2013 7:37:21 PM
-
-John Doe remarked: "Man, yeah.
-
-This is tricky stuff"
-==========
 Bogus Highlight
 nonsense
 

